include/nanobind/nb_lib.h

```diff
@@ -46,7 +46,7 @@ struct NB_CORE cleanup_list {
     void expand() noexcept;
 
 protected:
-    uint32_t m_size;
+    uint32_t m_size{1};
```

Comment:
This change hurts readability. There is only one constructor, and keeping the initialization there is preferable to spreading it out between multiple places.